game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/AaCasualtySelectorTest.java

```diff
@@ -417,4 +419,606 @@ void hitsLessThanPlanesKillsAccordingToTheRolledDice() {
+              "text",
+              // low luck would have 1 hit because 6 power / 6 dice side
+              givenDiceRollWithHitSequence(true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat(details.getKilled(), hasSize(1));
+      assertThat(
+          "The planes have 1 hp so there can't be damaged planes",
+          details.getDamaged(),
+          is(empty()));
+    }
+
+    @Test
+    void twoTypesOfPlanesAndBothHaveAmountEqualToDiceSides() {
+
+      final List<Unit> planes = planeUnitType.createTemp(6, hitPlayer);
+      planes.addAll(otherPlaneUnitType.createTemp(6, hitPlayer));
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              planes,
+              aaUnitType.createTemp(1, aaPlayer),
+              mock(CombatValue.class),
+              givenAaCombatValue(),
+              "text",
+              // low luck would have 2 hits because 12 power / 6 dice side
+              givenDiceRollWithHitSequence(true, true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat(details.getKilled(), hasSize(2));
+      assertThat(
+          "One of each plane type should be killed",
+          details.getKilled().stream().map(Unit::getType).collect(Collectors.toList()),
+          containsInAnyOrder(planeUnitType, otherPlaneUnitType));
+
+      assertThat(
+          "The planes have 1 hp so there can't be damaged planes",
+          details.getDamaged(),
+          is(empty()));
+    }
+
+    @Test
+    void twoTypesOfPlanesAndTogetherHaveAmountEqualToDiceSides() {
+
+      // need to randomly pick a plane to kill
+      whenGetRandom(bridge).thenAnswer(withValues(1));
+
+      final List<Unit> planes = planeUnitType.createTemp(3, hitPlayer);
+      planes.addAll(otherPlaneUnitType.createTemp(3, hitPlayer));
+
+      final CasualtyDetails details =
+          AaCasualtySelector.getAaCasualties(
+              planes,
+              aaUnitType.createTemp(1, aaPlayer),
+              mock(CombatValue.class),
+              givenAaCombatValue(),
+              "text",
+              // low luck would have 1 hits because 6 power / 6 dice side
+              givenDiceRollWithHitSequence(true),
+              bridge,
+              hitPlayer,
+              UUID.randomUUID(),
+              mock(Territory.class));
+
+      assertThat(details.getKilled(), hasSize(1));
+      assertThat(
+          "The otherPlaneUnitType is killed because it's unitType is sorted before planeUnitType "
+              + "(it sorts by unit type name) and there are 3 of them and the random selection was "
+              + "for the second unit in the list.",
+          details.getKilled().stream().map(Unit::getType).collect(Collectors.toList()),
+          containsInAnyOrder(otherPlaneUnitType));
+
+      assertThat(
```

Comment:
This has been asserted a few times, is there any way that exactly one of these tests would have this assertion fail and not the others, or is it more the case that if one fails they all would fail?

Reply:
Okay, a heuristic for good tests is that they are independent. This means if you make a change then you wouldn't break all tests, just the specific ones regarding what was changed (which should be relatively targeted).